includes/device-groups.inc.php

```diff
@@ -82,23 +83,29 @@ function GenGroupSQL($pattern, $search = '', $extra = 0)
     if ($pattern === false) {
         return false;
     }
-    $tmp    = explode(' ', $pattern);
-    $tables = array();
-    foreach ($tmp as $opt) {
-        if (strstr($opt, '%') && strstr($opt, '.')) {
-            $tmpp     = explode('.', $opt, 2);
-            $tmpp[0]  = str_replace('%', '', $tmpp[0]);
-            $tables[] = mres(str_replace('(', '', $tmpp[0]));
-            $pattern  = str_replace($opt, $tmpp[0].'.'.$tmpp[1], $pattern);
+
+    if (starts_with($pattern, '%')) {
+        // v1 pattern
+        $tables = array();
+        $words = explode(' ', $pattern);
+        foreach ($words as $word) {
+            if (starts_with($word, '%') && str_contains($word, '.')) {
+                list($table, $column) = explode('.', $word, 2);
+                $table = str_replace('%', '', $table);
+                $tables[] = mres(str_replace('(', '', $table));
+                $pattern = str_replace($word, $table . '.' . $column, $pattern);
+            }
         }
+        $tables = array_keys(array_flip($tables));
+    } else {
+        // v2 pattern
+        $tables = getTablesFromPattern($pattern);
     }
 
-    $pattern = rtrim($pattern, '&&');
-    $pattern = rtrim($pattern, '||');
+    $pattern = rtrim($pattern, '&|');
 
-    $tables = array_keys(array_flip($tables));
-    if (dbFetchCell('SELECT 1 FROM information_schema.COLUMNS WHERE TABLE_NAME = ? && COLUMN_NAME = ?', array($tables[0],'device_id')) != 1) {
-        //Our first table has no valid glue, append the 'devices' table to it!
+    if ($tables[0] != 'devices' && dbFetchCell('SELECT 1 FROM information_schema.COLUMNS WHERE TABLE_NAME = ? && COLUMN_NAME = ?', array($tables[0],'device_id')) != 1) {
```

Comment:
I don't get any device_group_device entries when running this now. I deleted them all to force it to re-cache and it seems to stop here as I have:

```bash
Array
(
    [0] => devices
)
Array
(
    [0] => devices
)
Array
(
    [0] => devices
)
Array
(
    [0] => devices
)
Array
(
    [0] => devices
)
```